test/unit/concepts/comparison/three_way_comparable.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Christopher Di Bella.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+#include "cjdb/concepts/comparison/three_way_comparable.hpp"
+
+#include <version>
+#if defined(__cpp_lib_three_way_comparison) and __cpp_lib_three_way_comparison >= 201711
+
+#include "cjdb/concepts/core/common_reference_with.hpp"
+#include "cjdb/concepts/core/convertible_to.hpp"
+#include "three_way_comparable.hpp"
+
+template<typename T, typename Cat, bool expected>
+constexpr void check_three_way_comparable_impl() noexcept
+{
+	static_assert(cjdb::three_way_comparable<T, Cat> == expected);
+}
+
+template<typename T, cjdb::common_reference_with<T> U, typename Cat, bool expected>
+constexpr void check_three_way_comparable_with_impl() noexcept
+{
+	static_assert(cjdb::three_way_comparable_with<T, U, Cat> == expected);
+}
```

Comment:
Also like... do you need these two functions? They're _barely_ shorter (and not shorter if you drop the `== true` and replace `== false` with `!`) and it's easier to follow if you just use the named concepts:

```cpp
check_three_way_comparable_impl<T, std::partial_ordering, true>();
static_assert(three_way_comparable<T, std::partial_ordering> == true);
static_assert(three_way_comparable<T, std::partial_ordering>);

check_three_way_comparable_with_impl<T, T, std::partial_ordering, true>();
static_assert(three_way_comparable_with<T, T, std::partial_ordering> == true);
static_assert(three_way_comparable_with<T, T, std::partial_ordering>);
```

Reply:
Yeah, good point. These functions are a relic of an older test, where I checked cv-qualified and ref-qualified stuff too.

Reply:
Applied in cf7734b02373a2a2bfbfaaa3753692cb89aad6ac.